### DISCUSSION TITLE
How do WPF Style Trigger animations work in WinUI?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/WindowsAppSDK#1297] [Q&A] [answered]
I have a WPF app that uses Style Triggers to start/stop animations.  WindowsAppSDK claims on the main Readme.md file that I don't have to change any of my code:

![image](https://user-images.githubusercontent.com/87879060/130535885-5df8c536-5add-44b1-86f2-767a810cddda.png)

Is this true? Does WinUI support both WPF Style Triggers and UWP VisualStateManager for animations?

**Comment by mdtauk** [comment, marked as answer]
The slight misunderstanding here is that, you would continue to use your WPF code base, and opt into the modern APIs and features that the Windows App SDK introduces.

WinUI would require at least a re-write or port of your WPF Xaml to WinUI's version of Xaml.

At some point it should be possible to use WinUI within a XAML Island for your existing WPF UI.

**Reply by ghost** [reply]
Thanks for the clarification, @mdtauk!  Very much appreciated.
